analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Sonar/HashesShouldHaveUnpredictableSalt.CSharp11.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Security.Cryptography;
+using System.Text;
+
+public class Sample
+{
+    public void Examples()
+    {
+        const string passwordString = "Secret";
+        var passwordBytes = Encoding.UTF8.GetBytes("Secret");
+
+        var shortSalt = "123456789012345"u8.ToArray();
+
+        var safeSalt = "1234567890123456"u8.ToArray();
+        RandomNumberGenerator.Create().GetNonZeroBytes(safeSalt);
+
+        using var pdb1 = new PasswordDeriveBytes(passwordBytes, shortSalt); // FN
```

Comment:
I think 'shortSalt' is misleading; if I understood correctly, it would fail in this case due to predictable salt (e.g. {{Make this salt unpredictable.}}) and not due to its shortness (e.g. {{Make this salt at least 16 bytes.}}). Maybe changing the name and having the same initialization as safeSalt could be better, or adding another test case for the actual FN to test the shortness.
```
 using var pdb1 = new PasswordDeriveBytes(passwordBytes, shortSalt); // FN
 RandomNumberGenerator.Create().GetNonZeroBytes(shortSalt);
 using var pdb2 = new PasswordDeriveBytes(passwordBytes, shortSalt); // FN
 using var pdb3 = new PasswordDeriveBytes(passwordBytes, safeSalt);
```

Reply:
It is just the way the message is written, but as you see the compliant case just below is also a hardcoded value but is 16 characters long and it is not raising.